.github/workflows/code_formatting_checks.yml

```diff
@@ -0,0 +1,55 @@
+name: Code formatting
+
+# Scheduled at 11:30 PM UTC (05:00 AM IST)
+on:
+  schedule:
+    - cron: "30 23 * * *"
+
+jobs:
+  fix_code_formatting:
+    runs-on: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Fix code formatting issues
+        run: ./gradlew spotlessApply
+
+      - name: Commit and Push
+        run: ./scripts/commit_and_push.sh "${{secrets.USER_NAME}}" "${{secrets.USER_EMAIL}}"
+
+      - name: Create PR
+        uses: actions/github-script@0.3.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+            var date = new Date();
+            var monthName = monthNames[date.getMonth()];
+            var day = `${date.getDate()}`;
+            if (day.length < 2) {
+                day = `0${date.getDate()}`;
+            }
+            day = `${monthName}-${day}-${date.getFullYear()}`
+            var branchName = `ga/${day}/code-formatting`
```

Comment:
We are depending on an implementation detail that the `commit_and_push` script sets the branch to this specific name. If we change the branch formatting there, we will also have to remember to change this here. I think we can set outputs from a step and then consume that as inputs in other steps (https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#outputs).

Can we look at this to pass the created branch name to the `create PR` step?

Reply:
At the moment GitHub script action doesn't allow taking input. One possible way is writing our own javascript action that takes input and uses in the script

https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-a-javascript-action

Reply:
I think we might be able to use it. Anyway, we can merge this in now and check back on this later, not a blocker to merge.